AK/String.h

```diff
@@ -30,7 +30,7 @@ class StringData;
 }
 
 // FIXME: Remove this when Apple Clang and OpenBSD Clang fully supports consteval.
-#if defined(AK_OS_MACOS) || defined(AK_OS_OPENBSD)
+#if defined(AK_OS_MACOS) || defined(AK_OS_OPENBSD) || (defined(__clang__) && __clang_major__ < 15)
```

Comment:
Let's check `defined(OSS_FUZZ)` here, and then I think we're good to merge this to fix the oss-fuzz pipeline. Not too concerned about clang < 15 tbh.

It's set here: 

https://github.com/SerenityOS/serenity/blob/50413c23266926944c641a99bcd03461e5224a8b/Meta/Lagom/BuildFuzzers.sh#L72

Reply:
(with a FIXME, tho)